src/dodal/beamlines/i22.py

```diff
@@ -23,27 +29,81 @@
 from dodal.log import set_beamline as set_log_beamline
 from dodal.utils import BeamlinePrefix, get_beamline_name, skip_device
 
-BL = get_beamline_name("i22")
+LAB_NAME = BEAMLINE_LAB_MAPPING.get("i22", "i22")
+
+BL = LAB_NAME if LAB_FLAG else get_beamline_name("i22")
+print("BL NAME: ", BL)
 set_log_beamline(BL)
 set_utils_beamline(BL)
 
+IS_LAB = LAB_FLAG and BL == LAB_NAME
+print("is this lab? : ", IS_LAB)
+
```

Comment:
Revert most of this: current behaviour is as expected: if $BEAMLINE is set to p38, we get that otherwise we get i22. 

IS_LAB can stay, but without LAB_FLAG. We should always try and connect to whatever $BEAMLINE is set to.

Reply:
the `--lab` was suggested by @callumforrester and then implemented. I find it neater as is emphasizes the link between the i22 and p38

Reply:
I suggested a much more granular approach. Rather than a global `--lab` flag (which just sounds like GDA dummy mode, and is likely to have all the same problems), have as many "profiles" as you like per beamline.

What @DiamondJoseph suggests is much closer to the approach currently used in other parts of dodal at the moment, so I suggest we stick with that and revisit the profiles approach as a separate ticket (which, to be honest, is what I thought you were doing).

Reply:
> I have no idea what dodal profiles are....

They are a thing I made up, they do not actually exist, don't worry :)

Reply:
pushed the revert